### ISSUE TITLE
GCM click_action is not sending

### ISSUE THREAD

**Issue by usckuro** [google/gcm#180]
Actually I'm using GCM to send all the push notifications to android and iOS devices, but when I add the click_action property it doesn't stick to the push, but the other properties appears.

Notification received:

```
 {
   aps =     {
       alert =         {
           body = “Lorem Ipsum…..";
           title = “New Message";
       };
       badge = 1;
       content-available = 1;
       sound = default;
   };
}
```

this is one example of how I build the payload

```
$payLoad['notification'] = [
                'title' => $push->getTitle(),
                'text'  => $push->getAlert(),
                'badge' => 1,
                'sound' => 'default',
                'click_action' => $push->getClickAction()
];
```

**Comment by mohitsharma0690** [comment]
What's the value you're setting for click_action. It accepts a "string". Does the same push for android also drop click_action or does it have it?

**Comment by usckuro** [comment]
Yes it is a string. I also hardcoded the value but that even doesn't work

**Comment by mohitsharma0690** [comment]
Is it only dropped for iOS or also for Android?

**Comment by usckuro** [comment]
both of them :/

**Comment by ogeidix** [comment]
@usckuro I checked and "click_action" is working correctly on my test app on Android.
Note that I sent the message using a simple curl command line, I didn't use the PHP code in your example.

I suggest to check your client implementation, or the sending library that you are using.
